internals/models.py

```diff
@@ -885,6 +885,46 @@ def getSortingMilestone(feature):
 
     return allowed_feature_list
 
+  @classmethod
+  def get_in_milestone(
+      self, version=None, show_unlisted=False, milestone=None):
+    if milestone == None:
+      return None
+
+    logging.info('Getting chronological feature list in milestone %d', milestone)
+    q = Feature.query()
+    q = q.order(Feature.name)
+    q = q.filter(Feature.shipped_milestone == milestone)
+    desktop_shipping_features = q.fetch(None)
+
+    # Features with an android shipping milestone but no desktop milestone.
+    q = Feature.query()
+    q = q.order(Feature.name)
+    q = q.filter(Feature.shipped_android_milestone == milestone)
+    q = q.filter(Feature.shipped_milestone == None)
+    android_only_shipping_features = q.fetch(None)
+
+    # Constructor the proper ordering.
+    all_features = []
+    all_features.extend(desktop_shipping_features)
+    all_features.extend(android_only_shipping_features)
+
+    # First sort by name, then sort by feature milestone (latest first).
+    all_features.sort(key=lambda f: f.name, reverse=False)
+
+    all_features = [f for f in all_features if not f.deleted]
+
+    feature_list = [f.format_for_template(version) for f in all_features]
+
+    self._annotate_first_of_milestones(feature_list, version=version)
```

Comment:
I think that this will be needed.

Reply:
The `get_chronological()` method returns the list of all features such that features having similar implementation statuses are listed together. Note that the features that have shipped milestone field or shipped android milestone, are grouped together according to the chromium version number instead of the implementation status.

Hence, features with implementation statuses as `'Behind a flag'`, `'Enabled by default'`, `'Deprecated'`, `'Removed'`, `'Origin trial'` etc., which always have shipped milestone field or shipped android milestone, are always grouped according to the chromium version numbers.

It seems that `_annotate_first_of_milestones()` annotates the first feature of each implementation statuse (which IS NOT associated with version numbers) and each chromium version. It fails to annotate the first feature which is `behind the flag`, `removed`, `deprecated` etc. 

Since, all the features that are returned by the `get_in_milestone()` method belong to a certain version number and hence has one of the implementation statuses from `'Behind a flag'`, `'Enabled by default'`, `'Deprecated'`, `'Removed'`, `'Origin trial'` etc., the method `_annotate_first_of_milestones()` will not be of any use to us.

Reply:
I did not notice it earlier. To sort the features in the given chromium version according to the implementation status, we will have to use a new method.

Reply:
See [this](https://github.com/GoogleChrome/chromium-dashboard/pull/1424/commits/8f84d1147cd56a3adb440b35663c7b4f635c64bc) commit

Reply:
Good point.  That looks good now.

Reply:
Okay. Marking this conversation as resolved.